src/types.ts

```diff
@@ -31,9 +31,18 @@ export type CSSFuncProps<T> = {
     | ((point: T, index: number, data: T[]) => CSSProperties[key])
     | CSSProperties[key];
 };
+export type CSSFuncPropsId = {
+  [key in keyof CSSProperties]:
+    | ((id: string) => CSSProperties[key])
+    | CSSProperties[key];
+};
+export type OnClickProp = (
+  result?: React.MouseEvent<SVGGElement, MouseEvent>,
+) => void;
+export type OnClickFuncProps = ((id?: string) => OnClickProp) | OnClickProp;
```

Comment:
I don't see a reason for the `(id?: string) => OnClickProp` version. `onClick` could directly receive two arguments (the id and the event).

Reply:
yes it would be more simple
done